### DISCUSSION TITLE
How to specify `packageManager` major version instead of specific one in package.json?

### DISCUSSION THREAD

**Discussion by amatlash** [pnpm/pnpm#7961] [Q&A] [answered]
In my package.json I have `"packageManager": "^pnpm@9.0.0"`
After using `pnpm install` for example i'm getting this error:
`ERR_PNPM_OTHER_PM_EXPECTED  This project is configured to use ^pnpm`

I've also tried `pnpm@9.0.0`, `pnpm@^9.0.0` - not working 

What is the correct way to specify major version?
And is it correct way to deal with package versions?

**Comment by zkochan** [comment, marked as answer]
It only works with exact versions. This is how it was designed by corepack team. If you want to ignore the error, set package-manager-strict to `false`: https://pnpm.io/npmrc#package-manager-strict

Or just remove the field.

**Comment by hlovdal** [comment]
> pnpm@^9.0.0 - not working

That should be [`^pnpm@9.0.0`](https://stackoverflow.com/a/72825528/23118).

**Reply by zkochan** [reply]
No, the packageManager only works with exact versions. What you posted is not valid. pnpm v11 will also work with devEngines.packageManager, which allows ranges.
